sdk/servicebus/azure-messaging-servicebus/src/samples/java/com/azure/messaging/servicebus/ReceiveMessageAndExtendLockSample.java

```diff
@@ -0,0 +1,130 @@
+    private final String messageTrackingUUID = UUID.randomUUID().toString();
+    private final DirectProcessor<Message> messageProcessor = DirectProcessor.create();
+    private final DirectProcessor<Throwable> errorProcessor = DirectProcessor.create();
+    private final DirectProcessor<AmqpEndpointState> endpointProcessor = DirectProcessor.create();
+    private final DirectProcessor<AmqpShutdownSignal> shutdownProcessor = DirectProcessor.create();
+
+    @Mock
+    private AmqpReceiveLink amqpReceiveLink;
+
+    @Captor
+    private ArgumentCaptor<Supplier<Integer>> creditSupplier;
+
+    private Mono<AmqpReceiveLink> receiveLinkMono;
+    private List<Message> messages = new ArrayList<>();
+
+    private ServiceBusReceiverAsyncClient consumer;
+
+
+    @BeforeEach
+    public void setup() {
+        MockitoAnnotations.initMocks(this);
+        receiveLinkMono = Mono.fromCallable(() -> amqpReceiveLink);
+
+        when(amqpReceiveLink.receive()).thenReturn(messageProcessor);
+        //when(amqpReceiveLink.getErrors()).thenReturn(errorProcessor);
+        //when(amqpReceiveLink.getConnectionStates()).thenReturn(endpointProcessor);
+        //when(amqpReceiveLink.getShutdownSignals()).thenReturn(shutdownProcessor);
+
+        String connectionString = System.getenv("AZURE_SERVICEBUS_CONNECTION_STRING")
+            + ";EntityPath=hemant-test1";
+
+        // Instantiate a client that will be used to call the service.
+
+        consumer = new ServiceBusClientBuilder()
+            .connectionString(connectionString)
+            .scheduler(Schedulers.elastic())
+            .buildAsyncReceiverClient();
+    }
+
+    @AfterEach
+    public void teardown() {
+        messages.clear();
+        Mockito.framework().clearInlineMocks();
+        consumer.close();
+    }
+
+    @Test
+    public void receiveAndExtendLockMessage() {
+
+        // Arrange
+        final int numberOfEvents = 1;
+
+        // Act & Assert
+        StepVerifier.create(
+            consumer.receive()
+                .take(numberOfEvents))
+            .assertNext(receivedMessage -> {
+                AtomicReference<Instant> timeToRefresh = new AtomicReference<>(receivedMessage.getLockedUntil());
+                log(" Got message time to refresh in " + receivedMessage.getLockedUntil());
+                Disposable renewDisposable = consumer.renewMessageLock(receivedMessage)
+                    .repeat(() -> true)
+                    .delayElements(Duration.ofSeconds(1))
+                    .subscribe(instant -> {
+                        log(" New time instant:" + instant);
+                        timeToRefresh.set(instant);
+                    });
+
+                // processing the messaging
+                int count = 0;
+                while (count < 15) {
+                    ++count;
+                    log(count + ". processing message ");
+                    try {
+                        Thread.sleep(1000);
```

Comment:
We shouldn't be using Thread.sleep() to test or wait. StepVerifier was created so it could wrap the async operations and wait correctly.